openbb_platform/providers/biztoc/openbb_biztoc/models/global_news.py

```diff
@@ -0,0 +1,164 @@
+    ]
+):
+    """Transform the query, extract and transform the data from the Biztoc endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> BiztocGlobalNewsQueryParams:
+        """Transform the query."""
+        return BiztocGlobalNewsQueryParams(**params)
+
+    @staticmethod
+    def get_news(
+        api_key: str,
+        filter: Literal[
+            "crypto", "hot", "latest", "main", "media", "source", "tag"
+        ] = "latest",
+        source: str = "bloomberg",
+        tag: str = "",
+        term: str = "",
+    ) -> List[Dict]:
+        """Calls the BizToc API and returns the data."""
+
+        term = term.replace(" ", "%20") if term else ""
+        _tags = get_all_tags(api_key)
+        pages = get_pages(api_key)
+        tags = []
+        tag = tag.lower() if tag else ""
+        for page in pages:
+            tags.extend(_tags[page][:])
+
+        _sources = get_sources(api_key)
+        sources = sorted([i["id"] for i in _sources])
+
+        headers = {
+            "X-RapidAPI-Key": f"{api_key}",
+            "X-RapidAPI-Host": "biztoc.p.rapidapi.com",
+            "Accept": "application/json",
+            "Accept-Encoding": "gzip",
+        }
+
+        filter_dict = {
+            "hot": "news/hot",
+            "latest": "news/latest",
+            "crypto": "news/latest/crypto",
+            "main": "news/latest/main",
+            "media": "news/latest/media",
+            "source": f"news/source/{source.lower()}",
+            "tag": f"tag/{tag}",
+        }
+        if filter == "source" and source.lower() not in sources:
+            raise ValueError(f"{source} not a valid source. Valid sources: {sources}")
+
+        if filter == "tag" and tag.lower().replace(" ", "") not in tags:
+            raise ValueError(f"{tag} not a valid tag. Valid tags: {tags}")
+
+        url = (
+            f"https://biztoc.p.rapidapi.com/search?q={term}"
+            if term
+            else f"https://biztoc.p.rapidapi.com/{filter_dict[filter]}"
+        )
+        r = make_request(url, headers=headers)
+        if r.status_code != 200:
+            raise RuntimeError(f"HTTP error - > {r.text}")
+
+        return r.json()
+
+    @staticmethod
+    def extract_data(
+        query: BiztocGlobalNewsQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Extract the data from the Biztoc endpoint."""
+
+        api_key = credentials.get("biztoc_api_key") if credentials else ""
+
+        data = BiztocGlobalNewsFetcher.get_news(
+            api_key=api_key, filter=query.filter, source=query.source, tag=query.tag, term=query.term  # type: ignore
+        )
+        if query.filter == "hot":
+            _data = []
+            for i in range(0, len(data)):
+                _posts = data[i]["posts"]
+                _data.extend(_posts)
+            data = _data
+
+        times = {"2 Hours Ago": 2, "4 Hours Ago": 4}
```

Comment:
This should be done in `extract_data` function.

Reply:
Which is it, move it to `helpers` or move it into `extract_data`?  Can't have it both ways.

Reply:
Move `get_news` to `helpers` and the coder after this comment to `extract_data`

Reply:
This *is* the `extract_data` function.